Core/GDCore/IDE/Events/UsedExtensionsFinder.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * GDevelop Core
+ * Copyright 2008-2016 Florian Rival (Florian.Rival@gmail.com). All rights
+ * reserved. This project is released under the MIT License.
+ */
+
+#ifndef GDCORE_USED_EXTENSIONS_FINDER_H
+#define GDCORE_USED_EXTENSIONS_FINDER_H
+#include <set>
+
+#include "GDCore/Events/Parsers/ExpressionParser2NodeWorker.h"
+#include "GDCore/IDE/Events/ArbitraryEventsWorker.h"
+#include "GDCore/IDE/Project/ArbitraryObjectsWorker.h"
+#include "GDCore/String.h"
+
+namespace gd {
+class Project;
+class Object;
+class BehaviorContent;
+}  // namespace gd
+
+namespace gd {
+
+class GD_CORE_API UsedExtensionsFinder
+    : public ArbitraryObjectsWorker,
+      public ArbitraryEventsWorkerWithContext,
+      public ExpressionParser2NodeWorker {
+ public:
+  static std::set<gd::String> ScanProject(gd::Project& project);
+
+ private:
+  UsedExtensionsFinder(gd::Project& _project) { projectPtr = &_project; };
+  gd::Project* projectPtr;
```

Comment:
nitpicking: if you don't need to change the pointer and it's always defined, you can use a reference, that you initialize in the initializer list of the constructor:
```suggestion
  UsedExtensionsFinder(gd::Project& project_): project(project_) {};
  gd::Project& project;
```

References are safer because the compiler enforce their initialization. Also uber nit: use an underscore at the end of the name. Underscores at the beginning are either for unused parameters (though in C++ you can omit their name) or implementation specific stuff (usually variables starting with double underscore).

Reply:
I originally used a reference but the compiler complained that I wasn't defining it in the constructor even though I was